table/block_based_table_reader.h

```diff
@@ -488,11 +493,12 @@ struct BlockBasedTable::Rep {
 
   // Footer contains the fixed table information
   Footer footer;
-  // index_reader and filter will be populated and used only when
-  // options.block_cache is nullptr; otherwise we will get the index block via
-  // the block cache.
+  // index_reader, filter, and uncompression_dict will be populated and used
+  // only when options.block_cache is nullptr; otherwise we will get the index
+  // block via the block cache.
   std::unique_ptr<IndexReader> index_reader;
   std::unique_ptr<FilterBlockReader> filter;
+  std::unique_ptr<UncompressionDict> uncompression_dict;
```

Comment:
Probably explicitly document the ownership of `index_reader`, `filter`, and `uncompression_dict` in different cases?

Reply:
Sure, will try.